packages/cardano-services/src/Projection/createTypeormProjection.ts

```diff
@@ -114,6 +117,10 @@ export const createTypeormProjection = ({
     migrationsRun: true
   });
   return projectionSource$.pipe(
+    tap((event) => {
+      // eslint-disable-next-line unicorn/no-process-exit
+      if (exitAtBlockNo && event.block.header.blockNo >= exitAtBlockNo) process.exit(0);
+    }),
```

Comment:
It would be cleaner to stop the projection here (or, probably better, **after** projecting this last block) instead of process exit, then calling process.exit in the new `finalize()` operator in `ProjectionHttpService.ts`.

Also, it is more efficient to apply the operator conditionally, rather than checking `exitAtBlockNo` for every block.
```suggestion
    exitAtLastBlockNo ? takeWhile((event) => event.block.header.blockNo < exitAtBlockNo) : passthrough(),
```

Reply:
`// eslint-disable-next-line unicorn/no-process-exit`

Is a smell 😄

Reply:
dc9f6ccd4 should solve this